packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-transform.ts

```diff
@@ -0,0 +1,252 @@
+    this.resourceTemplateObj.addCfnCondition(
+      {
+        expression: cdk.Fn.conditionEquals(cdk.Fn.ref('env'), 'NONE'),
+      },
+      'ShouldNotCreateEnvResources',
+    );
+
+    // Add outputs
+    this.resourceTemplateObj.addCfnOutput(
+      {
+        value: cdk.Fn.join('', [
+          'https://',
+          cdk.Fn.ref(this.cliInputsState.resourceName),
+          '.execute-api.',
+          cdk.Fn.ref('AWS::Region'),
+          '.amazonaws.com/',
+          cdk.Fn.conditionIf('ShouldNotCreateEnvResources', 'Prod', cdk.Fn.ref('env')) as unknown as string,
+        ]),
+        description: 'Root URL of the API gateway',
+      },
+      'RootUrl',
+    );
+
+    this.resourceTemplateObj.addCfnOutput(
+      {
+        value: this.resourceName,
+        description: 'API Friendly name',
+      },
+      'ApiName',
+    );
+
+    this.resourceTemplateObj.addCfnOutput(
+      {
+        value: cdk.Fn.ref(this.resourceName),
+        description: 'API ID (prefix of API URL)',
+      },
+      'ApiId',
+    );
+
+    // Add resources
+    this.resourceName === ADMIN_QUERIES_NAME
+      ? this.resourceTemplateObj.generateAdminQueriesStack(this.resourceName, authResourceName)
+      : this.resourceTemplateObj.generateStackResources(this.resourceName);
+  }
+
+  async applyOverrides() {
+    const backendDir = pathManager.getBackendDirPath();
+    const overrideFilePath = pathManager.getResourceDirectoryPath(undefined, AmplifyCategories.API, this.resourceName);
+    const overrideJSFilePath = path.join(overrideFilePath, 'build', 'override.js');
+
+    const isBuild = await buildOverrideDir(backendDir, overrideFilePath).catch(error => {
+      printer.debug(`Skipping build as ${error.message}`);
+      return false;
+    });
+
+    // skip if packageManager or override.ts not found
+    if (isBuild) {
+      const { override } = await import(overrideJSFilePath).catch(error => {
+        formatter.list(['No override file found', `To override ${this.resourceName} run "amplify override api"`]);
+        return undefined;
+      });
+
+      if (override && typeof override === 'function') {
+        const overrideCode: string = await fs.readFile(overrideJSFilePath, 'utf-8').catch(() => {
+          formatter.list(['No override File Found', `To override ${this.resourceName} run amplify override auth`]);
+          return '';
+        });
+
+        const sandboxNode = new vm.NodeVM({
+          console: 'inherit',
+          timeout: 5000,
+          sandbox: {},
+          require: {
+            context: 'sandbox',
+            builtin: ['path'],
+            external: true,
+          },
+        });
+
+        try {
+          await sandboxNode.run(overrideCode, overrideJSFilePath).override(this.resourceTemplateObj as AmplifyApigwResourceStack);
+        } catch (err: $TSAny) {
+          const error = new Error(`Skipping override due to ${err}.`);
+          printer.error(`${error}`);
+          error.stack = undefined;
```

Comment:
I would keep this stack. Would be useful for customers trying to debug their override files